src/main/java/uk/gov/companieshouse/filetransferservice/service/impl/AmazonFileTransferImpl.java

```diff
@@ -0,0 +1,301 @@
+
+        String bucket = getBucketFromS3Location(location);
+        String fileId = getKeyFromS3Location(location);
+
+        if (!s3Client.doesObjectExist(bucket, fileId))
+            throw new SdkClientException("S3 path does not exist: " + location);
+
+        return s3Client.getObject(new GetObjectRequest(bucket, fileId));
+    }
+
+
+    protected String getBucketFromS3Location(String location) {
+        String path = location.substring(SPLIT_S3_PATH_SUBSTRING);
+        return path.split("/")[0];
+    }
+
+    /**
+     * Get the fileId from the S3 location
+     */
+    protected String getKeyFromS3Location(String location) {
+        String path = location.substring(SPLIT_S3_PATH_SUBSTRING);
+        String[] pathSegments = path.split("/");
+
+        StringBuilder stringBuffer = new StringBuilder();
+        for (int i = 1; i < pathSegments.length - 1; i++) {
+            stringBuffer.append(pathSegments[i]);
+            stringBuffer.append('/');
+        }
+        stringBuffer.append(pathSegments[pathSegments.length - 1]);
+
+        return stringBuffer.toString();
+    }
+
+    /**
+     * Get the bucket name on S3
+     */
+    private String getAWSBucketName() {
+        return getSplitS3path()[0];
+    }
+
+    private String getPathIfExists() {
+        if (getSplitS3path().length > 1) {
+            return getSplitS3path()[1] + SUFFIX;
+        } else {
+            return "";
+        }
+    }
+
+    /**
+     * Get the S3 path split
+     */
+    private String[] getSplitS3path() {
+        String s3Path = configuration.getS3Path();
+        String path = s3Path.substring(SPLIT_S3_PATH_SUBSTRING);
+        return path.split("/", GET_S3_PATH_FROM_SPLIT);
+    }
+
+    /**
+     * Get the proxy host if it has been defined
+     *
+     * @return A {@link String} or null
+     */
+    private String getProxyHost() {
+        return System.getenv(PROXY_HOST_PROPERTY);
+    }
+
+    /**
+     * Get the proxy port if it has been defined
+     *
+     * @return An {@link Integer} or null
+     */
+    private Integer getProxyPort() {
+        String proxyPortString = System.getenv(PROXY_PORT_PROPERTY);
+        return proxyPortString == null ? null : Integer.valueOf(proxyPortString);
+    }
+
+    /**
+     * Create a S3 client
+     */
+    protected AmazonS3 getAmazonS3Client() {
+        AWSCredentials credentials = new BasicAWSCredentials(configuration.getAccessKeyId(),
+                configuration.getSecretAccessKey());
+        return getAmazonS3Client(credentials);
+    }
+
+    /**
+     * Create a S3 client
+     *
+     * @param credentials
+     * @return An {@link AmazonS3}
+     */
+    private AmazonS3 getAmazonS3Client(AWSCredentials credentials) {
+        ClientConfiguration clientConfiguration = getClientConfiguration(getProxyHost(),
+                getProxyPort(),
+                configuration.getProtocol());
+        return new AmazonS3Client(credentials, clientConfiguration);
+    }
```

Comment:
I think it would be better to inject the `AmazonS3` instance through dependency injection, and create it in a configuration class. This would aid in testability, and ensure that this class is only doing one thing and adhering to the single responsibility principle.

Reply:
done